core/src/test/java/org/hisp/dhis/android/core/period/internal/PeriodGeneratorCompatibilityShould.kt

```diff
@@ -0,0 +1,247 @@
+            52 * years,
+        ),
+        Generator(
+            WeeklyPeriodGeneratorFactory.sunday(calendar),
+            NewWeeklyPeriodGeneratorFactory.sunday(clock),
+            52 * years,
+        ),
+        Generator(
+            BiWeeklyPeriodGenerator(calendar),
+            NewBiWeeklyPeriodGenerator(clock),
+            12 * years,
+        ),
+        Generator(
+            MonthlyPeriodGenerator(calendar),
+            NewMonthlyPeriodGenerator(clock),
+            12 * years,
+        ),
+        Generator(
+            NMonthlyPeriodGeneratorFactory.biMonthly(calendar),
+            NewNMonthlyPeriodGeneratorFactory.biMonthly(clock),
+            6 * years,
+        ),
+        Generator(
+            NMonthlyPeriodGeneratorFactory.quarter(calendar),
+            NewNMonthlyPeriodGeneratorFactory.quarter(clock),
+            4 * years,
+        ),
+        Generator(
+            NMonthlyPeriodGeneratorFactory.sixMonthly(calendar),
+            NewNMonthlyPeriodGeneratorFactory.sixMonthly(clock),
+            2 * years,
+        ),
+        Generator(
+            NMonthlyPeriodGeneratorFactory.sixMonthlyApril(calendar),
+            NewNMonthlyPeriodGeneratorFactory.sixMonthlyApril(clock),
+            2 * years,
+        ),
+        Generator(
+            NMonthlyPeriodGeneratorFactory.sixMonthlyNov(calendar),
+            NewNMonthlyPeriodGeneratorFactory.sixMonthlyNov(clock),
+            2 * years,
+        ),
+        Generator(
+            YearlyPeriodGeneratorFactory.yearly(calendar),
+            NewYearlyPeriodGeneratorFactory.yearly(clock),
+            years,
+        ),
+        Generator(
+            YearlyPeriodGeneratorFactory.financialApril(calendar),
+            NewYearlyPeriodGeneratorFactory.financialApril(clock),
+            years,
+        ),
+        Generator(
+            YearlyPeriodGeneratorFactory.financialJuly(calendar),
+            NewYearlyPeriodGeneratorFactory.financialJuly(clock),
+            years,
+        ),
+        Generator(
+            YearlyPeriodGeneratorFactory.financialOct(calendar),
+            NewYearlyPeriodGeneratorFactory.financialOct(clock),
+            years,
+        ),
+        Generator(
+            YearlyPeriodGeneratorFactory.financialNov(calendar),
+            NewYearlyPeriodGeneratorFactory.financialNov(clock),
+            years,
+        ),
+    )
+
+    @Test
+    fun generate_periods() {
+        generators.forEach { generator ->
+            val oldPeriods = generator.oldGenerator.generatePeriods(-generator.periods, generator.periods)
+            val newPeriods = generator.newGenerator.generatePeriods(-generator.periods, generator.periods)
+
+            areEqual(oldPeriods, newPeriods)
+        }
+    }
+
+    @Test
+    fun generate_periods_in_year() {
+        (-5..5).forEach { offset ->
+            generators
+                .filterNot { it.newGenerator is YearlyPeriodGenerator }
```

Comment:
There are some problems in the old generator with Yearly generators periods starting in October and November. This method does not recognized correctly the current year for the given period type. As this method (`periodsInYear`) is not used with Yearly periods, YearlyGenerators are filtered out.